pkg/virt-handler/vm.go

```diff
@@ -217,6 +222,46 @@ func (d *VMHandlerDispatch) Execute(store cache.Store, queue workqueue.RateLimit
 	return
 }
 
+func (d *VMHandlerDispatch) mapNodeNetworkToLibvirt(vm *v1.VirtualMachine) (*v1.VirtualMachine, error) {
+
+	var node *k8sv1.Node
+
+	for i, iface := range vm.Spec.Domain.Devices.Interfaces {
+		if iface.Type == "nodeNetwork" {
+			if node == nil {
+				var err error
+				node, err = d.clientset.CoreV1().Nodes().Get(vm.Status.NodeName, metav1.GetOptions{})
+				if err != nil {
+					return nil, err
+				}
+			}
+			nodeIP := networking.GetNodeInternalIP(node)
+			if nodeIP == "" {
+				return nil, fmt.Errorf("No Node IP detected.")
+			}
+			link, err := d.networkIntrospector.GetLinkByIP(nodeIP, "1")
+
+			if err != nil {
+				return nil, fmt.Errorf("Could not detect interface for IP %v: %v", nodeIP, err)
+			}
+
+			obj, err := model.Clone(&iface)
+			if err != nil {
+				return nil, err
+			}
+
+			newIf := obj.(*v1.Interface)
+			newIf.Type = "direct"
+			newIf.Source = v1.InterfaceSource{
+				Device: link.Name,
+				Mode:   "bridge",
+			}
+			vm.Spec.Domain.Devices.Interfaces[i] = *newIf
```

Comment:
This change to the VM spec seems like it could conflict with migrations.  If the VM is migrated, the interface won't get re-detected because its been converted to type "direct".